### PR TITLE
HPCC-13533 Fix invalid code generated for disk normalize

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review.

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

File: ecl/hqlcpp/hqlsource.cpp [hunk 1]
```diff
@@ -3333,7 +3333,7 @@ void ChildAggregateBuilder::buildTransform(IHqlExpression * expr)
     BuildCtx transformCtx(instance->startctx);
     transformCtx.addQuotedCompound("virtual void processRows(ARowBuilder & crSelf)");
     translator.ensureRowAllocated(transformCtx, "crSelf");
-    buildTransformBody(transformCtx, expr, false, false);
+    buildTransformBody(transformCtx, expr, false, false, false);
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlsource.cpp]: Is this correct?

File: ecl/hqlcpp/hqlsource.cpp [hunk 2]
```diff
@@ -3396,7 +3396,7 @@ void ChildGroupAggregateBuilder::buildTransform(IHqlExpression * expr)
 {
     BuildCtx transformCtx(instance->startctx);
     transformCtx.addQuotedCompound("void processRows(IHThorGroupAggregateCallback * callback)");
-    buildGroupAggregateTransformBody(transformCtx, expr, true);
+    buildGroupAggregateTransformBody(transformCtx, expr, true, false);
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlsource.cpp]: Likewise?

**Comment by richardkchapman** [comment]
@ghalliday Please respond

File: ecl/hqlcpp/hqlsource.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlsource.cpp]: yes.  If you look at the prototype of the function that is being generated there is no parameter for the current input row - so it should not be bound.  Same goes for example below.

**Comment by ghalliday** [comment]
@richardkchapman see response.

**Comment by jamienoss** [comment]
@richardkchapman This is good to merge.

**Pull request merged**